src/test/java/com/microsoft/sqlserver/jdbc/connection/ConnectionTest.java

```diff
@@ -43,4 +43,32 @@ public void testConnections() throws SQLException {
         try (Connection con = ds.getConnection()) {
         }
     }
+    
+    @Test
+    public void testConnectWithIPAddressPreference () throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        ds.setURL(connectionString);
+        ds.setIPAddressPreference("IPv4First");
+        
+        try (Connection con = ds.getConnection()) {
+        }
+        
+        ds.setIPAddressPreference("IPv6First");
+        
+        try (Connection con = ds.getConnection()) {
+        }
+        
+        ds.setIPAddressPreference("UsePlatformDefault");
+        
+        try (Connection con = ds.getConnection()) {
+        }
+        
+        ds.setIPAddressPreference("Bogus");
+        

```

Comment:
there should be a R_expectedFailPassed here

Reply:
Agreed, changes made.